models/client/reducers.js

```diff
@@ -0,0 +1,48 @@
+module.exports = globalConfig => ({
+  updateUsername: updateUsername,
+  connectivityChange: connectivityChange(globalConfig),
+  suggestRecovery: suggestRecovery,
+  denyRecovery: denyRecovery,
+  activateRecovery: activateRecovery(globalConfig)
+})
+
+function connectivityChange (globalConfig) {
+  return inner
+  function inner (state, info) {
+    state.group = info.GID
+    state.id = info.CID
+    state.connectivityState = info.state
+    return state
+  }
+}
+
+function updateUsername (state, name) {
+  state.username = name
+  return state
+}
+
+function activateRecovery(globalConfig) {
+  return inner
+  function inner(state, _) {
```

Comment:
I've seen this convention in other programming languages. The problem is that we are using [lodash](https://lodash.com/) a utility library which uses _ as namespace similar to the $ of jQuery.